### ISSUE TITLE
Add Unit tests and CI to project?

### ISSUE THREAD

**Issue by marigerr** [chingu-voyage-turtles-team-5/momentum-clone#18]
I think it would be fun and good practice to set up a couple tests and use the github integration with a CI tool like Travis.

Might help with our reviewing of PRs as well especially if we work on one main js file together, might be more chance for an unintended side effect.

**Comment by jrpcoder** [comment]
Yes, it is great practice to do unit testing in our work. I approve that. I'll have a look into Travis.

**Comment by jrpcoder** [comment]
Ok, been looking at Travis. Isn't it only usable with node.js?

**Comment by marigerr** [comment]
I've never set it up myself, but I believe node.js is just used to run the tests locally and on Travis.  The project doesn't need to include a backend.
Here is a little blog about using Karma/Jasmine with Travis.   
https://www.sitepoint.com/testing-javascript-jasmine-travis-karma/
I think it might be doable to add a couple simple tests.  I'd be happy to try it out and see if it's not too much of a hassle.   
I also found this project, but maybe little more advanced than we have time for:
https://github.com/acvetkov/sinon-chrome

**Comment by jrpcoder** [comment]
Ok, its doable. I'll have a look at those too. I would like to implement unit testing as well. I think it is very important and as you said it would be great practice and ideal before merging pull requests.

**Comment by marigerr** [comment]
I've been trying a few different testing suites out.  One challenge is I believe most of our tests will be GUI related and/or testing ajax calls to APIs rather than testing more simple javascript functions.   
I think I'm personally going to focus a little less on getting testing setup and focus more on getting some of the features implemented.   But going to try to come back to this now and then, in between working on  other things.
